File: 0541-Reverse String II/0541-Reverse String II.cpp
class Solution {
public:
    string reverseStr(string s, int k) {
        for (int i = 0; i < s.size(); i += 2 * k)
        {
            int end = min(i + k, (int)(s.size()));
            reverse(s.begin() + i, s.begin() + end);
        }
        
        return s;
    }
};

File: 0501-0600/0541-Reverse String II/0541-Reverse String II.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        l = list(s)
        for i in range(0, len(l), k*2):
            l[i:i + k] = reversed(l[i:i + k])
        return ''.join(l)
